relay-general/src/pii/event_paths.rs

```diff
@@ -0,0 +1,213 @@
+use std::collections::BTreeSet;
+
+use crate::processor::{
+    process_value, Pii, ProcessValue, ProcessingState, Processor, SelectorPathItem, SelectorSpec,
+    ValueType,
+};
+use crate::protocol::{AsPair, Event, PairList};
+use crate::types::{Annotated, Meta, ProcessingResult};
+
+use crate::pii::processor::process_pairlist;
+
+struct EventPathsProcessor {
+    paths: BTreeSet<String>,
+}
+
+impl Processor for EventPathsProcessor {
+    fn before_process<T: ProcessValue>(
+        &mut self,
+        value: Option<&T>,
+        _meta: &mut Meta,
+        state: &ProcessingState<'_>,
+    ) -> ProcessingResult {
+        // The following skip-conditions are in sync with what the PiiProcessor does.
+        if state.value_type() == Some(ValueType::Boolean)
+            || value.is_none()
+            || state.attrs().pii == Pii::False
+        {
+            return Ok(());
+        }
+
+        macro_rules! insert_path {
```

Comment:
Can this `macro` be a function? It looks like this doesn't do anything funky and just might need a trait. I would prefer that strongly over a macro if possible.

Reply:
Agreed